EDF.py

```diff
@@ -8,32 +41,44 @@
 import re
 import warnings
 
+
 def padtrim(buf, num):
     num -= len(buf)
-    if num>=0:
-        # pad the input to the specified length
-        return str(buf) + ' ' * num
-    else:
-        # trim the input to the specified length
-        return buf[0:num]
-
-####################################################################################################
-# the EDF header is represented as a tuple of (meas_info, chan_info)
-# meas_info should have ['record_length', 'magic', 'hour', 'subject_id', 'recording_id', 'n_records', 'month', 'subtype', 'second', 'nchan', 'data_size', 'data_offset', 'lowpass', 'year', 'highpass', 'day', 'minute']
-# chan_info should have ['physical_min', 'transducers', 'physical_max', 'digital_max', 'ch_names', 'n_samps', 'units', 'digital_min']
-####################################################################################################
+    return (str(buf) + ' ' * num) if (num >= 0) else buf[0:num]
+
+
+def writebyte(file, content, encoding='utf8'):
+    try:
+        # Py3 onwards bytes and strings are separate data format
+        content = bytes(content, encoding)
+    except TypeError:
+        # Py2.7 Support
+        pass
+    except Exception as e:
+        print(type(e))
+        print(str(e))
+        print(
+            "If you see this message, please go to " + 
+            "https://github.com/bids-standard/pyedf/issues" + 
+            " and open an issue there regarding this. Thank you!")
+    finally:
+        file.write(content)
```

Comment:
```suggestion
    if type(content) != bytes:
        content = content.encode(encoding)
      
    file.write(content)
```

(This works in Python 2.7 and Python 3 -- I haven't tested it in older Python 2, but am not interesting in supporting that!)

Also, note that the `finally` clause here doesn't make much sense: if there has been a conversion error, then you don't want to try to write the bytes out to file. It would be better to simply leave out the extra `except Exception as e` and `finally` clauses and have `file.write` be after the `try` block. But in this case, there's an even easier solution (see suggestion).

Reply:
Since I implemented Error Handling block here, I used `finally` clause to help people understand the code flow. Since there is only one line left to do after the block, and I know developers who can indent such lines one level inside the `except` clause thinking it as a mistake on our end. I wish to avoid clearing up such confusion in the future if anyone faces this.

I added the `except Exception as e` clause to ensure people know what to do if they face such error. I can always do away with this. This is only there for least expected scenarios.

`bytes` is not a supported type in Python 2.7 so the `if` clause will not work there. However, as we have already added the support in `except TypeError` clause, how about we put the `content = content.encode(encoding)` line in there? This will ensure that encoding is done correctly for Python 2.7.

Reply:
```python
Python 2.7.17 (default, Jul 20 2020, 15:37:01) 
[GCC 7.5.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> x = "a"
>>> type(x)
<type 'str'>
>>> type(x) == bytes
True
>>> type(x) == str
True
```

Reply:
Right. `bytes` is a synonym for `str` in Python 2.7.

I mean, if we are to do `ascii` encoding either way, why not enforce the line in the `TypeError` block, which will be triggered in Python 2.7 since default `str` encoding scheme is `utf-8`.

Reply:
Okay, I ran a bit of code for testing the error handling block in Python 2.7 and I feel the block is necessary. Run the following example - 

```python
# -*- coding: utf-8 -*-
enc = 'ascii'
x = 'µ Õ ß'

print x

try:
    y = x.encode(enc)
except Exception as e:
    print("Error Message: " + str(e))
    print("Error Type: " + str(type(e)))
    y = 'fail'

print y
print (x == y)
 ```

The output will be like the following - 
```python
Python 2.7.16 (default, Jul 13 2019, 16:01:51)
[GCC 8.3.0] on linux2

µ Õ ß
Error Message: 'ascii' codec can't decode byte 0xc2 in position 0: ordinal not in range(128)
Error Type: <type 'exceptions.UnicodeDecodeError'>
fail
False
```

So errors will occur if there are non-ascii characters, and the code will fail. Now the question is if it is better to let the code fail, or if we should handle the error gracefully and display it to the console for the developer. Let me know your opinion about this @palday .

Reply:
@palday waiting for your response.